manifest/provider/apply.go

```diff
@@ -195,9 +200,10 @@ func (s *RawProviderServer) ApplyResourceChange(ctx context.Context, req *tfprot
 				}
 				return v, ap.NewError(err)
 			}
-			nv, err := morph.ValueToType(ppMan.(tftypes.Value), v.Type(), tftypes.NewAttributePath())
-			if err != nil {
-				return v, ap.NewError(err)
+			nv, d := morph.ValueToType(ppMan.(tftypes.Value), v.Type(), tftypes.NewAttributePath())
+			if len(d) > 0 {
+				resp.Diagnostics = append(resp.Diagnostics, d...)
```

Comment:
As with my other comment we should probably add some sort of actionable message that precedes the diagnostics from the morpher. 

Something like 

```
Could not determine the Terraform type information for the resource. This can happen when _______. 

See the diagnostics below for the affected fields. 
```

Reply:
i added a preceding diag hinting at schema issues and pointing at details of following errors.